Sources/RayTracer.Library/Serialization/Serializers/IntersectableListSerializer.cs

```diff
@@ -0,0 +1,56 @@
+using System;
+using System.Text.Json;
+using RayTracer.Library.Extensions;
+using RayTracer.Library.Shapes;
+
+namespace RayTracer.Library.Serialization.Serializers;
+
+public class IntersectableListSerializer : SerializerBase<IntersectableListSerializer, IntersectableList>
```

Comment:
looks like this can just be a `ProxySerializer`, reusing the logic for `ImmutableArraySerializer`